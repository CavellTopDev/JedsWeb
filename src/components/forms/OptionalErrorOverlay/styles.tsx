import { BORDER_RADIUS, COLORS, Z_INDEX_HIGHEST } from "../../../constants";

export const error = {
  background: COLORS.orange,
  borderRadius: BORDER_RADIUS,
  bottom: "-4px",
  boxShadow: "0 3px 6px -2px rgba(0, 0, 0, .1)",
  color: COLORS.white,
  fontSize: ".75em",
  fontWeight: 600,
  height: "20px",
  lineHeight: "20px",
  margin: 0,
  padding: "0 10px",
  position: "absolute" as "absolute",
  right: "-4px",
  zIndex: Z_INDEX_HIGHEST
};

export const container = {
  display: "block" as "block",
  position: "relative" as "relative"
};
