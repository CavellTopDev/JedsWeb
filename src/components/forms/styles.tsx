import {
  BORDER_RADIUS,
  BOX_SHADOW_BORDER,
  BOX_SHADOW_BORDER_DARKER,
  BOX_SHADOW_FOCUS,
  BOX_SHADOW_SITTING,
  COLORS
} from "../../constants";
import { styledViaProps } from "../../util/styled";

const DISABLED_OPACITY = 0.5;

export const ColoredButtonStyle = styledViaProps(
  "button",
  ({ color, disabled, fullWidth = false, height = 40 }) => JSON.stringify({ color, disabled, fullWidth, height }),
  ({
    color,
    disabled,
    fullWidth = false,
    height = 40
  }: {
    color: string;
    disabled: boolean;
    fullWidth?: boolean;
    height?: number;
  }) => ({
    backgroundColor: color,
    boxShadow: BOX_SHADOW_SITTING,
    color: COLORS.white,
    height: `${height}px`,
    lineHeight: `${height}px`,
    padding: "0 12px",
    pointerEvents: disabled ? "none" : undefined,
    opacity: disabled ? DISABLED_OPACITY : undefined,
    width: fullWidth ? "100%" : "auto",

    $nest: {
      "&:hover": {
        boxShadow: BOX_SHADOW_SITTING,
        opacity: disabled ? DISABLED_OPACITY : 0.95
      },
      "&:active": {
        opacity: disabled ? DISABLED_OPACITY : 1
      },
      "&:focus": {
        boxShadow: BOX_SHADOW_FOCUS
      }
    }
  })
);

export const DefaultButtonStyle = styledViaProps(
  "button",
  ({ disabled, fullWidth = false, height = 40 }) => JSON.stringify({ disabled, fullWidth, height }),
  ({ disabled, fullWidth = false, height = 40 }: { disabled: boolean; fullWidth?: boolean; height?: number }) => ({
    backgroundColor: COLORS.white,
    boxShadow: `${BOX_SHADOW_BORDER}, ${BOX_SHADOW_SITTING}`,
    color: COLORS.dp60,
    height: `${height}px`,
    lineHeight: `${height}px`,
    padding: "0 12px",
    pointerEvents: disabled ? "none" : undefined,
    opacity: disabled ? DISABLED_OPACITY : undefined,
    width: fullWidth ? "100%" : "auto",

    $nest: {
      "&:hover": {
        boxShadow: `${BOX_SHADOW_BORDER_DARKER}, ${BOX_SHADOW_SITTING}`,
        color: COLORS.darkpurple
      },
      "&:active": {
        backgroundColor: COLORS.dp02
      },
      "&:focus": {
        boxShadow: `${BOX_SHADOW_BORDER}, ${BOX_SHADOW_FOCUS}`
      }
    }
  })
);

export const SubtleButtonStyle = styledViaProps(
  "button",
  ({ fullWidth, height }) => JSON.stringify({ fullWidth, height }),
  ({ fullWidth = false, height = 32 }: { fullWidth?: boolean; height?: number }) => ({
    backgroundColor: "transparent",
    color: "inherit",
    height: `${height}px`,
    lineHeight: `${height}px`,
    maxWidth: "100%",
    padding: "0 8px",
    width: fullWidth ? "100%" : "auto",

    $nest: {
      "&:hover": {
        backgroundColor: "rgba(36, 18, 77, 0.04)",
        color: COLORS.purple
      },
      "&:active": {
        backgroundColor: "rgba(36, 18, 77, 0.08)",
        color: COLORS.purple
      },
      "&:focus": {
        boxShadow: BOX_SHADOW_FOCUS
      }
    }
  })
);

export const buttonBaseStyles = {
  borderRadius: BORDER_RADIUS,
  border: "none",
  cursor: "pointer",
  display: "inline-block" as "inline-block",
  fontSize: "14px",
  fontWeight: 500 as 500,
  margin: 0,
  outline: "none",
  overflow: "hidden" as "hidden",
  textOverflow: "ellipsis",
  verticalAlign: "middle",
  whiteSpace: "nowrap" as "nowrap"
};

export const formControl = {
  backgroundColor: COLORS.white,
  borderRadius: BORDER_RADIUS,
  border: "none",
  boxShadow: `${BOX_SHADOW_BORDER}, ${BOX_SHADOW_SITTING}`,
  boxSizing: "border-box" as "border-box",
  color: COLORS.darkpurple,
  fontSize: "14px",
  fontWeight: 500,
  height: "40px",
  lineHeight: "20px",
  outline: "none",
  padding: "0 12px",
  whiteSpace: "pre-wrap" as "pre-wrap",
  width: "100%",

  $nest: {
    "&:hover": {
      boxShadow: `${BOX_SHADOW_BORDER_DARKER}, ${BOX_SHADOW_SITTING}`
    },

    "&:focus": {
      boxShadow: `${BOX_SHADOW_BORDER}, ${BOX_SHADOW_FOCUS}`
    },

    "&::placeholder, &::-webkit-input-placeholder": {
      color: COLORS.dp40
    }
  }
};
