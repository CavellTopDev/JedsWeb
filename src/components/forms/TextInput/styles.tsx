import { formControl } from "../styles";

const baseStyles = [formControl];

export const normal = [...baseStyles];
export const error = [...baseStyles];
