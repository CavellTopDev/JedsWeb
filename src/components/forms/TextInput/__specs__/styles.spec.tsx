import "jest";
import * as styles from "../styles";

describe("TextInput styles", () => {
  describe("snapshots", () => {
    it("normal", () => {
      expect(styles.normal).toMatchSnapshot();
    });

    it("error", () => {
      expect(styles.error).toMatchSnapshot();
    });
  });
});
