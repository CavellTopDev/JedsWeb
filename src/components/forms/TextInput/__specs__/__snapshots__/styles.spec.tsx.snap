// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput styles snapshots error 1`] = `
Array [
  Object {
    "$nest": Object {
      "&::placeholder, &::-webkit-input-placeholder": Object {
        "color": "#a19dac",
      },
      "&:focus": Object {
        "boxShadow": "0 0 0 1px rgba(20, 11, 47, .1), 0 0 0 4px rgba(91, 147, 255, .25)",
      },
      "&:hover": Object {
        "boxShadow": "0 0 0 1px rgba(20, 11, 47, .15), 0 2px 4px -2px rgba(0, 0, 0, .2)",
      },
    },
    "backgroundColor": "#fff",
    "border": "none",
    "borderRadius": "3px",
    "boxShadow": "0 0 0 1px rgba(20, 11, 47, .1), 0 2px 4px -2px rgba(0, 0, 0, .2)",
    "boxSizing": "border-box",
    "color": "#24124d",
    "fontSize": "14px",
    "fontWeight": 500,
    "height": "40px",
    "lineHeight": "20px",
    "outline": "none",
    "padding": "0 12px",
    "whiteSpace": "pre-wrap",
    "width": "100%",
  },
]
`;

exports[`TextInput styles snapshots normal 1`] = `
Array [
  Object {
    "$nest": Object {
      "&::placeholder, &::-webkit-input-placeholder": Object {
        "color": "#a19dac",
      },
      "&:focus": Object {
        "boxShadow": "0 0 0 1px rgba(20, 11, 47, .1), 0 0 0 4px rgba(91, 147, 255, .25)",
      },
      "&:hover": Object {
        "boxShadow": "0 0 0 1px rgba(20, 11, 47, .15), 0 2px 4px -2px rgba(0, 0, 0, .2)",
      },
    },
    "backgroundColor": "#fff",
    "border": "none",
    "borderRadius": "3px",
    "boxShadow": "0 0 0 1px rgba(20, 11, 47, .1), 0 2px 4px -2px rgba(0, 0, 0, .2)",
    "boxSizing": "border-box",
    "color": "#24124d",
    "fontSize": "14px",
    "fontWeight": 500,
    "height": "40px",
    "lineHeight": "20px",
    "outline": "none",
    "padding": "0 12px",
    "whiteSpace": "pre-wrap",
    "width": "100%",
  },
]
`;
